nrf_modem/doc/delta-dfu.rst

```diff
@@ -0,0 +1,204 @@
+
+Limitations
+===========
+
+The size of the scratch area can be retrieved with the :c:func:`nrf_modem_delta_dfu_area` function, which returns the size of the area in bytes.
+A limitation of delta updates is that the size of the update may not exceed the area of FLASH memory that a given modem firmware version has dedicated for this operation.
+The size of this area varies for any given modem firmware version and it may be retrieved with the modem library :c:func:`nrf_modem_delta_dfu_area` function.
+In general, this function can be useful when the size of the largest delta update that can be received needs to be reported to an external service, for example an LwM2M server.
+In pratice, the application does not use this function in the normal update process, since all delta update files produced by Nordic Semiconductor are guaranteed to fit in the scratch area of the firmware version that they apply to.
+
+Features
+********
+
+Security
+========
+
+Modem firmware delta updates implement security features to ensure that only firmware updates coming from Nordic Semiconductor are ever programmed to the modem.
+Delta update files are encrypted and signed. The modem takes care of veryfing the signature and decrypting the content as necessary.
+
+Fail-safety
+===========
+
+Modem firmware delta updates implement several failsafe mechanisms to ensure that the modem continues to operate as normal in case of errors.
+
+#. If the modem were to detect issues with the firmware being transfered, such as the firmware being corrupt, the modem automatically aborts the firmware update process and continues to operate as normal.
+#. If the modem were to detect issues with the new firmware during the update process at boot time, such as the inability to communicate with the application via the modem library, it will automatically rollback to the previous firmware version and continue to operate as normal.
+#. If the device were to suddenly lose power during the transfer of the new firmware, the application can resume the operation when power is restored and the modem is booted again.
+#. If the device were to suddenly lose power during the update process at boot time, the modem resumes the operation automatically when it is booted again.
+
+Firmware rollback
+=================
+
+The modem rollbacks to the previous version automatically in case it detects problems during boot. In case no problems are detected by the modem during boot, the modem will run the new firmware.
+If the application detects issues with the new modem firmware, it can manually rollback to the previous modem firmware by calling :c:func:`nrf_modem_delta_dfu_rollback`.
+The operation is scheduled, and the modem rollbacks to the previous modem firmware version the next time it is booted. Because the same firmware delta update
```

Comment:
Looks like this paragraph is unfinished?

Reply:
Fixed